pkg/monitor/controller.go

```diff
@@ -128,15 +129,16 @@ func (monitor *controllerMonitor) Run(stopCh <-chan struct{}) {
 		return
 	}
 
-	monitor.deleteStaleAgentCRDs()
-
 	// Sync controller monitoring CRD every minute util stopCh is closed.
 	go wait.Until(monitor.syncControllerCRD, time.Minute, stopCh)
 
-	for i := 0; i < defaultWorkers; i++ {
-		go wait.Until(monitor.nodeWorker, time.Second, stopCh)
-		if monitor.externalNodeEnabled {
-			go wait.Until(monitor.externalNodeWorker, time.Second, stopCh)
+	if monitor.antreaAgentInfoAPIAvailable() {
```

Comment:
We should log the result for visibility and troubleshooting. For example:
```
if monitor.antreaAgentInfoAPIAvailable() {
    klog.InfoS("The AntreaAgentInfo API is available, running node workers")
    monitor.deleteStaleAgentCRDs()
    ...
} else {
    klog.InfoS("The AntreaAgentInfo API is unavailable, will not run node workers")
}
```

Reply:
done.